src/schema-manager/scripted-http-schema-manager.ts

```diff
@@ -0,0 +1,63 @@
+import { SchemaManagerHost } from './types';
+import { RequestSetup, isRequestSetup } from './request-introspection-query';
+import { join } from 'path';
+import { HttpSchemaManager } from './http-schema-manager';
+
+interface ScriptedHttpSchemaManagerOptions {
+  fromScript: string;
+}
+
+export class ScriptedHttpSchemaManager extends HttpSchemaManager {
+  private _scriptFile: string;
+
+  constructor(_host: SchemaManagerHost, options: ScriptedHttpSchemaManagerOptions) {
+    super(_host);
+    this._scriptFile = options.fromScript;
+  }
+
+  private _requireScript(path: string) {
+    return require(path);
```

Comment:
```suggestion
    delete require.cache[path];
    return require(path);
```

User configuration script might be changed during TypeScript lang server lives. 

The `_getOptions()` method can be called several times via `HttpSchemaManager.startWatch` and  
the  `_requireScript` should return the latest version configuration script.

Reply:
I'm not sure how it all works together, so I have a question: is this ([here](https://github.com/thebartekbanach/ts-graphql-plugin/blob/e16896f1ef5c3d9f62382ecaf41ab174d9ec2e92/src/schema-manager/scripted-http-schema-manager.ts#L23-L25) and [here](https://github.com/thebartekbanach/ts-graphql-plugin/blob/e16896f1ef5c3d9f62382ecaf41ab174d9ec2e92/src/schema-manager/scripted-http-schema-manager.ts#L60), I don't know if there is a better way to reference to code in comment) "caching layer" a problem? Because even if I will clear require cache, the configuration will be still stored in `this._options` property and returned from it.

Reply:
Good catch, I missed `_options` in my review.

HttpSchemaManager and ScriptedHttpSchemaManager should not have store request options. Please remove `this._options = ...` .

Reply:
After a longer reflection I don't think if it is good idea to require and execute configuration script every second (if I correctly understand what you want me to do). It may be not optimal, because it would need to read script file from the file system, parse and execute it every second. It would also cause additional overhead for server, that would need, for example, to create new auth token per every second.

So my idea is to:
- reload script only after user change, then execute it again
- execute script again after schema fetch error (auth token may expire during development)

What do you think about it?
Can I watch for script file change using `SchemaManagerHost.watchFile(...)` method?

Reply:
Your idea looks good.

> Can I watch for script file change using SchemaManagerHost.watchFile(...) method?

Yes, you can register callback function like this:

```ts
    this._host.watchFile(join(this._host.getProjectRootPath(), this._scriptFile), () => this._options = null, 100);
```

And we need to override `startWatch` method in order to `execute script again after schema fetch error (auth token may expire during development)` because HttpSchemaManager.startWatch does not accept callback when request is failed.

Reply:
I did it. Now it automatically reloads configuration if user change the configuration script and reloads configuration if any fetch error is ocurred. But I did it in other way than you proposed, because it needed less testing (😅) and prevented me from copying almost all of the code from startWatch method. 😄